dia-batching-server/src/api/custom/arsb.rs

```diff
@@ -0,0 +1,73 @@
+use crate::api::binance::BinancePriceApi;
+use crate::api::custom::AssetCompatibility;
+use crate::api::error::CustomError;
+use crate::types::{AssetSpecifier, Quotation};
+use async_trait::async_trait;
+use chrono::Utc;
+use rust_decimal::prelude::Zero;
+use rust_decimal::Decimal;
+
+// The blockchain and symbol for the Argentinian blue dollar.
+// These are the expected values for the asset specifier.
+pub const BLOCKCHAIN: &'static str = "FIAT";
+pub const SYMBOL: &'static str = "ARS-USD";
+
+/// Returns the price for the Argentinian blue dollar.
+/// The price is fetched from Binance as binance has a very accurate price for the blue dollar.
+pub struct ArsBluePriceView;
+
+#[async_trait]
+impl AssetCompatibility for ArsBluePriceView {
+	fn supports(&self, asset: &AssetSpecifier) -> bool {
+		asset.blockchain.to_uppercase() == BLOCKCHAIN && asset.symbol.to_uppercase() == SYMBOL
+	}
+
+	async fn get_price(&self, _asset: &AssetSpecifier) -> Result<Quotation, CustomError> {
+		ArsBluePriceView::get_price().await
+	}
+}
+
+impl ArsBluePriceView {
+	async fn get_price() -> Result<Quotation, CustomError> {
+		// Only the symbol is needed to get the price with the BinancePriceApi client.
+		let asset =
+			AssetSpecifier { blockchain: "Binance".to_string(), symbol: "USDTARS".to_string() };
+
+		let binance_price_api = BinancePriceApi::new();
+		// The direction of the conversion is ARS -> USDT
+		let usdt_ars_price = binance_price_api
+			.get_price(&asset)
```

Comment:
I find it unexpected that we return a whole `Quotation` here, then only select the `price` entry and then construct a new `Quotation` around that (multiplicative inverse of the) price.

Reply:
Right, but IMO there is no perfect solution here. 

I implemented it like this because: 
- The `BinancePriceApi` is now consistent with the other `CoingeckoPriceApi` and `PolygonPriceApi` which both return `Quotation`s from their `get_prices()` functions
- The `BinancePriceApi` is somewhat generic and not tightly coupled to the custom `ArsBluePriceView`

If I change the return type of `BinancePriceApi::get_price()` to `Result<Decimal, BinanceError>` then it would be inconsistent with the other price apis.

The second `Quotation` here is necessary because the `symbol` of the `AssetSpecifier` we pass to the `BinancePriceApi` needs to match the binance symbol ("USDTARS") but the quotation we expect on-chain needs to have the symbol `ARS-USD`. So we don't only need to inverse the price of the received binance quotation but also change `blockchain` and `symbol`.